seedsource_core/django/seedsource/management/commands/calculate_zone_transfers.py

```diff
@@ -7,268 +7,369 @@
+            default=None,
+            help="Comma delimited list of zones sets to analyze. (default is to analyze all available zone sets)",
+        )
 
-        if x_slice.stop - x_slice.start < 2:
-            x_slice = slice(x_slice.start, x_slice.start + 2)
-        if y_slice.stop - y_slice.start < 2:
-            y_slice = slice(y_slice.start, y_slice.start + 2)
+        parser.add_argument(
+            "--variables",
+            dest="variables",
+            default=None,
+            help="Comma delimited list of variables analyze. (default is to analyze all available variables)",
+        )
 
-        return elevation[y_slice, x_slice], data[y_slice, x_slice], coords.slice_by_window(Window(y_slice, x_slice))
+        parser.add_argument(
+            "--clear",
+            dest="clear",
+            action="store_true",
+            default=False,
+            help="If True, will automatically delete all transfer limits before calculating new ones",
+        )
 
-    def _write_limit(self, variable, time_period, zone, masked_data, low=None, high=None, label=None):
-        min_value = numpy.nanmin(masked_data)
-        max_value = numpy.nanmax(masked_data)
+    def _write_limit(
+        self, variable, time_period, zone, masked_data, band, elevation_service
+    ):
+        min_value = masked_data.min()
+        max_value = masked_data.max()
         transfer = (max_value - min_value) / 2.0
         center = max_value - transfer
 
-        if numpy.isnan(transfer) or hasattr(transfer, 'mask'):
-            self.stdout.write('WARNING: Transfer limit is NaN for {}, zone {}, band {}-{}'.format(
-                zone.zone_source.name, zone.zone_id, low, high
-            ))
-            return
+        low, high = band[:2]
+        label = ""
+        if len(band) > 2:
+            label = band[2]
 
         tl = TransferLimit.objects.create(
-            variable=variable, time_period=time_period, zone=zone, transfer=transfer, center=center, low=low,
-            high=high, label=label
+            variable=variable,
+            time_period=time_period,
+            zone=zone,
+            transfer=transfer,
+            center=center,
+            low=low,
+            high=high,
+            label=label,
+            elevation=elevation_service,
         )
 
-        transfers_by_variable = self.transfers_by_source.get(zone.zone_source.name, {})
-        transfers_by_variable[variable] = transfers_by_variable.get(variable, []) + [transfer]
-        self.transfers_by_source[zone.zone_source.name] = transfers_by_variable
+        # # store stransfer limits so that we can calculate averages
+        # transfers_by_variable = self.transfers_by_source.get(zone.zone_source.name, {})
+        # transfers_by_variable[variable] = transfers_by_variable.get(variable, []) + [
+        #     transfer
+        # ]
+        # self.transfers_by_source[zone.zone_source.name] = transfers_by_variable
 
-        return tl
+        # return tl
```

Comment:
I'm not sure what's going on here. If this code no longer needed? If so, why is it left in comments?

Reply:
Forgot to delete it while refactoring.  Removed.